components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -961,6 +961,59 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             ))).filter(e => e !== undefined) as WhitelistedRepository[];
     }
 
+    public async getSuggestedContextURLs(ctx: TraceContext): Promise<string[]> {
+        const user = this.checkUser("getSuggestedContextURLs");
+        const suggestions: string[] = [];
+
+        // Fetch all data sources in parallel for maximum speed (don't await before `Promise.allSettled(promises)` below!)
+        const promises = [];
+
+        // Example repositories
+        promises.push(this.getFeaturedRepositories(ctx).then(exampleRepos => {
+            // log('got example repos', exampleRepos);
+            exampleRepos.forEach(r => suggestions.push(r.url));
+        }));
+
+        // User repositories
+        user.identities.forEach(identity => {
+            const provider = {
+                'Public-GitLab': 'gitlab.com',
```

Comment:
should this not just go through all my providers (i.e. incl. any gitlab self-managed installations)